design_docs/generalize_calendar.md

```diff
@@ -0,0 +1,249 @@
+<h1> Title: Generalize Calendar supported by Analysis <br>
+Xylar Asay-Davis <br>
+date: 2017/02/09 <br>
+</h1>
+<h2> Summary </h2>
+Currently, the time variable in `xarray` data sets within MPAS-Analysis have two
+major shortcomings, inherited from `xarray` (through `pandas` and `numpy.datetime64`).
+First, only the Gregorian calendar is supported.  Second, there is not support
+for dates outside the years 1678 to 2262.  The analysis needs to support both
+the Gregorain  ('gregorian') and the 365-day ('gregorian_noleap') calendars.  It also needs to
+support, at a minimum, years between 0001 and 9999, and preferably arbitrary
+years both positive and negative.
+
+A major challenge is that it seems that xarray cannot easily be forced to
+use an alternative representation of dates to the troublesome
+`numpy.datetime64` type.  The most obvious alternative, `datetime.datetime`,
+seemingly cannot be useddirectly in `xarray` because objects of this type
+are converted to `numpy.datetime64` objects at various stages when using 
+features from pandas, raising errors when dates are out of range.  While an
+alternative date class (e.g. `netcdftime.DatetimNoLeap`) might be used to
+represent dates on the 'gregorian_noleap' calendar, there is no such
+preexisting alternative for the 'gregorian' calendar.
+
+The solution proposed herein is to store time as floating-point days since the
+reference date 0001-01-01 and to convert dates in this format to 
+`datetime.datetime` and `MpasRelativeDelta` objects whenever mathematical
+manipulation of dates is required.
```

Comment:
This is reasonable because the smallest unit of time needing to be accounted for is ms.  Note, for clarity indicate that 

```python
>>> import sys
>>> sys.float_info.epsilon*24*60*60*1000
1.9184653865522705e-08
>>> sys.float_info.max/365.
4.925186670855659e+305
```
which indicates we can easily accommodate ms as the finest unit of time as well as long time periods corresponding to paleoclimate timescales.  Note, we *cannot* support ns, but for climate scales we presently have no use case or notion of a use case where we need ns support.